src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.CoreRT.cs

```diff
@@ -0,0 +1,384 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Threading;
+using System.Runtime.CompilerServices;
+using Internal.Runtime.CompilerServices;
+
+namespace System.Runtime.InteropServices
+{
+    /// <summary>
+    /// Enumeration of flags for <see cref="ComWrappers.GetOrCreateComInterfaceForObject(object, CreateComInterfaceFlags)"/>.
+    /// </summary>
+    [Flags]
+    public enum CreateComInterfaceFlags
+    {
+        None = 0,
+
+        /// <summary>
+        /// The caller will provide an IUnknown Vtable.
+        /// </summary>
+        /// <remarks>
+        /// This is useful in scenarios when the caller has no need to rely on an IUnknown instance
+        /// that is used when running managed code is not possible (i.e. during a GC). In traditional
+        /// COM scenarios this is common, but scenarios involving <see href="https://docs.microsoft.com/windows/win32/api/windows.ui.xaml.hosting.referencetracker/nn-windows-ui-xaml-hosting-referencetracker-ireferencetrackertarget">Reference Tracker hosting</see>
+        /// calling of the IUnknown API during a GC is possible.
+        /// </remarks>
+        CallerDefinedIUnknown = 1,
+
+        /// <summary>
+        /// Flag used to indicate the COM interface should implement <see href="https://docs.microsoft.com/windows/win32/api/windows.ui.xaml.hosting.referencetracker/nn-windows-ui-xaml-hosting-referencetracker-ireferencetrackertarget">IReferenceTrackerTarget</see>.
+        /// When this flag is passed, the resulting COM interface will have an internal implementation of IUnknown
+        /// and as such none should be supplied by the caller.
+        /// </summary>
+        TrackerSupport = 2,
+    }
+
+    /// <summary>
+    /// Enumeration of flags for <see cref="ComWrappers.GetOrCreateObjectForComInstance(IntPtr, CreateObjectFlags)"/>.
+    /// </summary>
+    [Flags]
+    public enum CreateObjectFlags
+    {
+        None = 0,
+
+        /// <summary>
+        /// Indicate if the supplied external COM object implements the <see href="https://docs.microsoft.com/windows/win32/api/windows.ui.xaml.hosting.referencetracker/nn-windows-ui-xaml-hosting-referencetracker-ireferencetracker">IReferenceTracker</see>.
+        /// </summary>
+        TrackerObject = 1,
+
+        /// <summary>
+        /// Ignore any internal caching and always create a unique instance.
+        /// </summary>
+        UniqueInstance = 2,
```

Comment:
There are more values in this enum now. It would be nice to create a new file `src\System\Runtime\InteropServices\ComWrappers.cs` with definition of these enums and the parts of `ComWrappers.ComInterfaceEntry` and `ComWrappers.ComInterfaceDispatch` that are same between all implementations.

Reply:
Should file `ComWrappers.cs` be located in same folder as `ComWrappers.CoreRT.cs` right now?

Reply:
The shared part should be in `src\libraries\System.Private.CoreLib\src\System\Runtime\InteropServices\ComWrappers.cs`.

Ideally, this refactoring would be submitted to dotnet/runtime directly.

Reply:
See https://github.com/dotnet/runtime/pull/47982

Reply:
Since refactoring landed in dotnet/runtime, what's my next steps I can wait while you or Michal update from upstream, or I can do something like
```
git switch -c runtime-merge
git merge upstream/runtime-master
```

and then submit PR from that branch?

Reply:
I have submitted the merge in #657

Reply:
#657 has been merged.